### ISSUE TITLE
About ignore files handling 

### ISSUE THREAD

**Issue by diverger** [sublimehq/sublime_merge#1451]
Assume I have a folder structure as below:

L:.
│  .gitignore
│  1.txt
│
├─sub1
│  │  2.txt
│  │
│  └─SUB2
│          3.txt
│
└─sub2

If I want to ignore the contents under 'sub2xxxx', I add  'sub2*/' into the .gitignore file. Every time I try to stage changes under 'SUB2', SM alert me the file is ignored. It seems SM will also try to filter the 'SUB2' folder under 'sub1' folder. 

1. Must I add '/' before the names to ignore the files/folder under the root of the repos?
2. The names in .ignore is case-insensitive, right?

**Comment by srbs** [comment]
> Every time I try to stage changes under 'SUB2', SM alert me the file is ignored.

Yes, this is how git itself works: https://git-scm.com/docs/gitignore#_pattern_format (answers to 1 & 2 are yes)
